custom/auth/ante/min_initial_deposit.go

```diff
@@ -0,0 +1,77 @@
+package ante
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	core "github.com/terra-money/core/types"
+)
+
+// MinInitialDeposit Decorator will check Initial Deposits for MsgSubmitProposal
+type MinInitialDepositDecorator struct {
+	govKeeper      GovKeeper
+	treasuryKeeper TreasuryKeeper
+}
+
+// NewMinInitialDeposit returns new min initial deposit decorator instance
+func NewMinInitialDepositDecorator(govKeeper GovKeeper, treasuryKeeper TreasuryKeeper) MinInitialDepositDecorator {
+	return MinInitialDepositDecorator{
+		govKeeper:      govKeeper,
+		treasuryKeeper: treasuryKeeper,
+	}
+}
+
+// IsMsgSubmitProposal checks whether the input msg is a MsgSubmitProposal
+func IsMsgSubmitProposal(msg sdk.Msg) bool {
+	_, ok := msg.(*govtypes.MsgSubmitProposal)
+	return ok
+}
+
+// HandleCheckMinInitialDeposit
+func HandleCheckMinInitialDeposit(ctx sdk.Context, msg sdk.Msg, govKeeper GovKeeper, treasuryKeeper TreasuryKeeper) (err error) {
+	submitPropMsg, ok := msg.(*govtypes.MsgSubmitProposal)
+	if !ok {
+		return fmt.Errorf("could not dereference msg as MsgSubmitProposal")
+	}
+
+	minInitialDepositRatio := treasuryKeeper.GetParams(ctx).MinInitialDepositRatio
```

Comment:
just a question like why don't we just use govKeeper.GetDepositParams(ctx).MinDeposit only.

So 1000000*2 can just be 2000000

Reply:
Isn't that only the deposit to get to voting phase?

Reply:
Is that param currently unused? I don't see it set when querying treasury or gov params via terrad?

Reply:
The "MinDeposit" itself is just for getting into voting phase but not for creating the proposal at all.

Reply:
<img width="803" alt="Screenshot 2023-02-02 at 19 20 09" src="https://user-images.githubusercontent.com/23058164/216323198-75e412d5-56c5-46c1-abfb-f8c1e10e4d83.png">

I can still query it here

Reply:
It would be cleaner by using just MinDeposit than having another params MinInitialDepositRatio and then do MinDeposit*MinInitialDepositRatio

Reply:
If we just use that, that would prevent people from getting a proposal up that others can deposit on to reach voting stage. From what I understand this param just is the barrier to get from deposit to voting stage for a proposal. It can still be created with e.g. 1000 LUNC deposit, it just can't be voted on.
So that shall be avoided to stop the spam-flooding in deposit phase I think.

Reply:
The new "factor" then would be the minimum share of the min deposit that has to be directly deposited when the prop is created.

Reply:
@StrathCole: Yes correct.

Reply:
@nghuyenthevinh2000:  I don't really get your question here. The MinDeposit is the parameter that controls, when a proposal goes to voting phase. The MinInitialDepositRatio is the part of the min deposit, that needs to be deposited alongside the proposal on tx broadcas (to get the proposal from mempool into deposit phase in the first place so to speak). Does this answer your question?

Reply:
ah, I see, just wonder what is the meaning of MinInitialDepositRatio

Reply:
Let's say MinDeposit is 1,000,000. MinInitialDepositRatio is 0.3
Then you have to pay 300,000 to create the proposal and after another 700,000 deposited (by whoever does) it goes to voting phase. Currently the 300,000 are not necessary, you can create with only 1,000.